### PR TITLE
chore: added trace detail tests

### PR DESCRIPTION
**Pull request by SagarRajput-7**
### Summary

- Test cases covered:
   - should render tracedetail
   - should render tracedetail events tab
   - should toggle slider - selected span details
   - should be able to selected another span and see its detail
   - focus on selected span and reset focus action

#### Screenshots

NA

### CODE REVIEW

File: frontend/src/mocks-server/__mockdata__/tracedetail.ts [hunk 1]
```diff
@@ -0,0 +1,2089 @@
+/* eslint-disable sonarjs/no-duplicate-string */
+export const traceDetailResponse = [
```
Comment (SagarRajput-7) [inline frontend/src/mocks-server/__mockdata__/tracedetail.ts:2]: this is a big mock response as I tried but couldn't shorten this, as response struct is interconnect to each other

File: frontend/src/pages/TraceDetail/__test__/TraceDetail.test.tsx [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+		[
+			{ trace: 'HTTP GET /dispatch', duration: '776.76 ms', count: '50' },
+			{ trace: 'HTTP GET: /customer', duration: '349.44 ms', count: '4' },
+			{
+				trace: '/driver.DriverService/FindNearest',
+				duration: '173.10 ms',
+				count: '15',
+			},
+			// and so on ...
+		].forEach((traceDetail) => {
```
Comment (ahmadshaheer) [inline frontend/src/pages/TraceDetail/__test__/TraceDetail.test.tsx:63]: nit: can we move this to a variable?

File: frontend/src/pages/TraceDetail/__test__/TraceDetail.test.tsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+		[
+			{ title: 'client-uuid', value: '64a18ffd5f8adbfb' },
+			{ title: 'component', value: 'net/http' },
+			{ title: 'host.name', value: '4f6ec470feea' },
+			{ title: 'http.method', value: 'GET' },
+			{ title: 'http.url', value: '/route?dropoff=728%2C326&pickup=165%2C543' },
+			{ title: 'http.status_code', value: '200' },
+			{ title: 'ip', value: '172.25.0.2' },
+			{ title: 'opencensus.exporterversion', value: 'Jaeger-Go-2.30.0' },
+		].forEach((tag) => {
```
Comment (ahmadshaheer) [inline frontend/src/pages/TraceDetail/__test__/TraceDetail.test.tsx:94]: nit: can we move this to a variable?

File: frontend/src/pages/TraceDetail/__test__/TraceDetail.test.tsx [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+		fireEvent.click(getByText('Events'));
+
+		expect(await screen.findByText('HTTP request received')).toBeInTheDocument();
+
+		// event details
+		[
+			{ title: 'Event Start Time', value: '527.60 ms' },
+			{ title: 'level', value: 'info' },
+		].forEach((tag) => {
```
Comment (ahmadshaheer) [inline frontend/src/pages/TraceDetail/__test__/TraceDetail.test.tsx:122]: nit: can we move this to a variable?

File: frontend/src/pages/TraceDetail/__test__/TraceDetail.test.tsx [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+		const spanTitle = getByText('/driver.DriverService/FindNearest');
+		expect(spanTitle).toBeInTheDocument();
+		fireEvent.click(spanTitle);
+
+		// Tag details
+		[
+			{ title: 'client-uuid', value: '6fb81b8ca91b2b4d' },
+			{ title: 'component', value: 'gRPC' },
+			{ title: 'host.name', value: '4f6ec470feea' },
+		].forEach((tag) => {
```
Comment (ahmadshaheer) [inline frontend/src/pages/TraceDetail/__test__/TraceDetail.test.tsx:170]: nit: can we move this to a variable?

**Review by ahmadshaheer — DISMISSED** [review]

**Review by vikrantgupta25 — APPROVED** [review]
lgtm

**Review by YounixM — APPROVED** [review]

**Pull request merged**